indico/web/client/js/react/components/WTFOccurrencesField.jsx

```diff
@@ -22,19 +23,30 @@ export default function WTFOccurrencesField({fieldId, uses24HourFormat}) {
   const field = useMemo(() => document.getElementById(fieldId), [fieldId]);
   const [values, setValues] = useState(() =>
     JSON.parse(field.value).map(value => ({
-      date: moment(value.date, 'YYYY-MM-DD'),
+      date: moment(value.date, moment.HTML5_FMT.DATE),
       duration: value.duration,
       time: moment(value.time, 'HH:mm'),
       id: _.uniqueId(),
     }))
   );
 
+  useEffect(() => {
+    // we cannot do this directly in updateOccurrences since after adding a new occurrence it
+    // doesn't exist in the DOM yet
+    const dateInputs = field.parentElement.querySelectorAll('ind-date-picker > input');
+    values.forEach(({date}, i) => {
+      // time and duration don't allow entering invalid data without the field being in an invalid
+      // state, so no need for custom validation there
+      dateInputs[i].setCustomValidity(date?.isValid() ? '' : Translate.string('Invalid date'));
```

Comment:
Let's use the same error message everywhere? DateField uses `The entered date is invalid.`